File: src/main/Java/com/futureclient/gui/ClickGui.java
package com.futureclient.gui;

import com.futureclient.FutureClient;
import com.futureclient.api.Category;
import com.futureclient.api.Module;
import com.futureclient.util.RenderUtil;
import net.minecraft.client.gui.GuiScreen;
import java.awt.Color;
import java.io.IOException;

public class ClickGui extends GuiScreen {
    private int panelWidth = 100;
    private int panelHeight = 18;
    private int gap = 110;

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        this.drawDefaultBackground();
        
        int x = 20;
        int y = 20;

        for (Category cat : Category.values()) {
            // Draw Category Header
            RenderUtil.drawRect(x, y, x + panelWidth, y + panelHeight, new Color(50, 50, 50, 255).getRGB());
            mc.fontRendererObj.drawStringWithShadow(cat.name(), x + 5, y + 5, -1);

            int moduleY = y + panelHeight;
            
            for (Module m : FutureClient.instance.moduleManager.getModulesByCategory(cat)) {
                boolean hovered = isHovered(x, moduleY, panelWidth, panelHeight, mouseX, mouseY);
                int color = m.isToggled() ? new Color(0, 150, 0, 200).getRGB() : new Color(0, 0, 0, 180).getRGB();
                if (hovered) color = new Color(100, 100, 100, 200).getRGB();

                RenderUtil.drawRect(x, moduleY, x + panelWidth, moduleY + panelHeight, color);
                mc.fontRendererObj.drawString(m.getName(), x + 5, moduleY + 5, -1);
                
                moduleY += panelHeight;
            }
            x += gap;
        }
        
        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    @Override
    protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
        int x = 20;
        int y = 20;

        for (Category cat : Category.values()) {
            int moduleY = y + panelHeight;
            
            for (Module m : FutureClient.instance.moduleManager.getModulesByCategory(cat)) {
                if (isHovered(x, moduleY, panelWidth, panelHeight, mouseX, mouseY)) {
                    if (mouseButton == 0) { // Left Click
                        m.toggle();
                        FutureClient.instance.configManager.save(); // Save on change
                    }
                }
                moduleY += panelHeight;
            }
            x += gap;
        }
        super.mouseClicked(mouseX, mouseY, mouseButton);
    }

    private boolean isHovered(int x, int y, int width, int height, int mouseX, int mouseY) {
        return mouseX >= x && mouseX <= x + width && mouseY >= y && mouseY <= y + height;
    }
    
    @Override
    public boolean doesGuiPauseGame() {
        return false;
    }
}

File: src/main/kotlin/com/futureclient/impl/gui/click/ClickGui.kt
package com.futureclient.impl.gui.click

import com.futureclient.FutureClient
import com.futureclient.api.module.Category
import com.futureclient.util.RenderUtil
import net.minecraft.client.gui.GuiScreen
import java.awt.Color

class ClickGui : GuiScreen() {
    private val panels = ArrayList<Panel>()

    init {
        var x = 20
        for (cat in Category.values()) {
            panels.add(Panel(cat, x, 20))
            x += 110
        }
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        drawDefaultBackground()
        panels.forEach { it.draw(mouseX, mouseY) }
        super.drawScreen(mouseX, mouseY, partialTicks)
    }

    override fun mouseClicked(mouseX: Int, mouseY: Int, mouseButton: Int) {
        panels.forEach { it.mouseClicked(mouseX, mouseY, mouseButton) }
        super.mouseClicked(mouseX, mouseY, mouseButton)
    }
    
    override fun doesGuiPauseGame(): Boolean = false
}

class Panel(val category: Category, var x: Int, var y: Int) {
    private val width = 100
    private val height = 15
    private var open = true
    
    fun draw(mouseX: Int, mouseY: Int) {
        RenderUtil.drawRect(x, y, x + width, y + height, Color(0, 150, 255).rgb)
        FutureClient.mc.fontRendererObj.drawStringWithShadow(category.name, x + 4f, y + 4f, -1)

        if (open) {
            var yOffset = height
            val modules = FutureClient.instance.moduleManager.getModulesByCategory(category)
            
            for (mod in modules) {
                val color = if (mod.toggled) Color(40, 40, 40, 200).rgb else Color(20, 20, 20, 200).rgb
                RenderUtil.drawRect(x, y + yOffset, x + width, y + yOffset + height, color)
                FutureClient.mc.fontRendererObj.drawString(mod.name, x + 4, y + yOffset + 4, if(mod.toggled) -1 else Color.GRAY.rgb)
                yOffset += height
            }
            RenderUtil.drawBorderedRect(x, y, x + width, y + yOffset, 1, Color(0, 100, 200).rgb, 0)
        }
    }

    fun mouseClicked(mouseX: Int, mouseY: Int, button: Int) {
        if (isHovered(mouseX, mouseY, x, y, width, height)) {
            if (button == 1) open = !open
            return
        }
        if (open) {
            var yOffset = height
            val modules = FutureClient.instance.moduleManager.getModulesByCategory(category)
            for (mod in modules) {
                if (isHovered(mouseX, mouseY, x, y + yOffset, width, height)) {
                    if (button == 0) mod.toggle()
                }
                yOffset += height
            }
        }
    }

    private fun isHovered(mx: Int, my: Int, x: Int, y: Int, w: Int, h: Int): Boolean {
        return mx >= x && mx <= x + w && my >= y && my <= y + h
    }
}
